### ISSUE TITLE
Labels Not appearing correctly in tracklets, but fine in analyze videos

### ISSUE THREAD

**Issue by WrathUM** [DeepLabCut/DeepLabCut#991]
Thanks for opening this issue, and thanks for using DeepLabCut (we hope you are enjoying it ;). Please fill out the template completely, including the full "traceback" and input code that you ran to hit this error.

**Your Operating system and DeepLabCut version**

Please state your operating system, env, and which version of DeepLabCut you are using.
Example: Ubuntu 16.04 LTS, with an Anaconda Env, & DeepLabCut1.x or 2.x.

Please complete the following information about your system:

OS:  Windows 10
DeepLabCut Version: 2.2
Anaconda env used: GPU

I analyze the videos and that step works fine, however, afterwards I look at the tracklets GUI and one mouse is not labeled for certain parts of the video. Sometimes the labels appear in positions where the mouse isn't. 

[Please copy/paste the full terminal output of the error here!!!]

![image](https://user-images.githubusercontent.com/57931859/98744698-02c4fd00-2367-11eb-932d-2bf6c10ca0fa.png)

**How to Reproduce the problem**
1. Go to Refine Tracklets
2. Click on Launch GUI
3. Go through the frames

**Screenshots**
![image](https://user-images.githubusercontent.com/57931859/98744679-f771d180-2366-11eb-92d3-01a07640f690.png)

**Additional context**
I am not sure if this means that I loaded the wrong files, but I did click on the .pickle file and the corresponding video.

**Comment by jeylau** [comment]
Hey @WrathUM, what specific version of DLC are you using?

**Comment by WrathUM** [comment]
I am using 2.2b7 

Also, here are some more pictures of the labels sometimes being present and sometimes not. 

![image](https://user-images.githubusercontent.com/57931859/98860147-12e8e500-2418-11eb-8aee-9f3336f39f8c.png)

![image](https://user-images.githubusercontent.com/57931859/98860176-1e3c1080-2418-11eb-8592-3e12dc3e46d5.png)

![image](https://user-images.githubusercontent.com/57931859/98860195-272ce200-2418-11eb-8624-65c85494979a.png)

**Comment by MMathisLab** [comment]
Hi wrath, depends on the conf you set and the max gap to fill; first, please update to 2.2b8, there are several important bug fixes. Two, set max gap to fill to 0, see video here: https://www.youtube.com/watch?v=YSRQT8N2vFE

**Comment by WrathUM** [comment]
I followed the settings in the video and the problem still exists, although a little less. I am wondering if I just need to go through the refine labels and re-train it multiple times before the tracklets become much more stable and remain attached to the mouse instead of appearing at random places or not appearing at all.

**Comment by MMathisLab** [comment]
How does the full.mp4 look? Pose estimation and tracking are separate steps, ie pose estimation needs to be good first :)

**Comment by WrathUM** [comment]
![image](https://user-images.githubusercontent.com/57931859/99716177-a5af0280-2a5c-11eb-8efd-d8ac805789ed.png)

![image](https://user-images.githubusercontent.com/57931859/99716230-b5c6e200-2a5c-11eb-87d0-2c88488e2ed2.png)

![image](https://user-images.githubusercontent.com/57931859/99716268-bfe8e080-2a5c-11eb-8d44-bc98495c4dac.png)

The video files also have these problems. It seems that the tailend node is not appearing and some nodes in the rest of the skeleton appear here and there.

**Comment by MMathisLab** [comment]
Sorry, to be clear I mean the video that ends in full.mp4; I’m trying to understand how your pose estimation quality is. 

- how do the scoremaps look? RMSE? Etc? https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/maDLC_AdvUserGuide.md

See here: https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/maDLC_AdvUserGuide.md#evaluate-the-trained-network

And here: https://github.com/DeepLabCut/DeepLabCut/blob/master/docs/maDLC_AdvUserGuide.md#attention

**Comment by WrathUM** [comment]
Sorry for such a delayed response. I only just was able to get back to focusing on this work. I have found the full.mp4 files and included some pictures below. The problem of points not appearing on the mice still is present. 

This is true for both of the videos that I put into the DLC pipeline. 

![image](https://user-images.githubusercontent.com/57931859/103471566-b8447900-4d36-11eb-93f9-5029c2c87ed2.png)
![image](https://user-images.githubusercontent.com/57931859/103471586-e924ae00-4d36-11eb-9a43-f9c53ef371b3.png)

**Comment by MMathisLab** [comment]
This then means your pose estimator is not sufficient. You should (1) train with every part connected in the skeleton (see docs), and add more training data; how many images did you use? (closes #1076, as that is a duplicate)

**Comment by WrathUM** [comment]
I extracted an labeled 39 frames for each of the videos. I also clicked the skeleton connection GUI and made sure to connect the skeletons. 

Also to bump my question from 1076, is hardware a possible limiting factor?

**Comment by WrathUM** [comment]
I have been trying to perform the same process on my own personal laptop and I keep getting runtime errors everytime I close the GUI and try to re-open the GUI in the same session. i.e. I close the GUI for refine tracklets, which end up causing the following error.

![image](https://user-images.githubusercontent.com/57931859/104651904-2f4e0b80-566d-11eb-88d2-f48e69570a93.png)


I also am having difficulty using the lasso to re-assign the points to the opposite mice as in this simplified test set-up, I have one white mice and one black mice, which makes identifying them easy. However, the tracklets still seem to be flipping them at times.  However, the reduced, trimmed video file, which I extracted 40 frames out of the 60 seconds of footage does not seem to be losing the labels, only making jumps. 
![image](https://user-images.githubusercontent.com/57931859/104652269-b8fdd900-566d-11eb-90fa-08babaa27b18.png)


![image](https://user-images.githubusercontent.com/57931859/104652308-c6b35e80-566d-11eb-80e5-0c5ff90fc10a.png)

I made sure to synthesize all the feedback and suggestions that were given above and in other related posts into this attempt at labeling and producing tracklets, but these errors still remain. My laptop has an i7 cpu and a MX150 nvidia graphics card, just in case the computer specs would be a limiting factor here.
